TELCODOCS-2140 (#97359)
openshift/openshift-docs: hardware_accelerators/das-about-dynamic-accelerator-slicer-operator.adoc

:_mod-docs-content-type: ASSEMBLY
[id="das-about-dynamic-accelerator-slicer-operator"]
= Dynamic Accelerator Slicer (DAS) Operator 
include::_attributes/common-attributes.adoc[]
:context: das-about-dynamic-accelerator-slicer-operator

toc::[]

[IMPORTANT]
====

> Dynamic Accelerator Slicer Operator (DAS) is a Technology Preview feature and is subject to Technology Preview support limitations. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. 

Author: Replaced w/ the snippet.

For more information, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope]. 
====

The Dynamic Accelerator Slicer (DAS) Operator allows you to optimize GPU utilization for workloads on {product-title} through the use of dynamically slicing GPU accelerators in {product-title}, instead of relying on statically sliced GPUs defined when the node is booted. This allows you to dynamically slice GPUs based on specific workload demands, ensuring efficient resource utilization. 

Dynamic slicing is useful if you do not know all the accelerator partitions needed in advance on every node on the cluster.

> :FeatureName: Dynamic Accelerator Slicer Operator

Author: Moved this.

The DAS Operator currently includes a reference implementation for NVIDIA Multi-Instance GPU (MIG) and is designed to support additional technologies such as NVIDIA MPS or GPUs from other vendors in the future.

.Limitations

> Remember the following limitations when using the Dynamic Accelerator Slicer Operator:

Author: Made this change.

 * You need to identify potential incompatibilities and ensure the system works seamlessly with various GPU drivers and operating systems.

 * The Operator only works with specific MIG compatible NVIDIA GPUs and drivers, such as H100 and A100.

>  * The Operator cannot use only a subset of the GPUs of a node.

Author: No change.

 * The NVIDIA device plugin cannot be used together with the Dynamic Accelerator Slicer Operator to manage the GPU resources of a cluster.

[NOTE]
====

> The DAS Operator is designed to work with MIG-enabled GPUs. It allocates MIG slices instead of whole GPUs. Installing the DAS Operator prevents the use of the standard resource request through the NVIDIA device plugin such as `nvidia.com/gpu: "1"`, for allocating the entire GPU.

Author: No change.

This approach allows you to define static default time-slicing configurations, but it can lead to resource wastage if the slicing does not align with the actual workload demands. Additionally, changing static partitions created at node boot-time requires the eviction of all the workloads at the node level in order to create a new set of partitions. 

The Dynamic Accelerator Slicer (DAS) Operator allows you to optimize GPU utilization for workloads on {product-title} through the use of dynamically slicing GPU accelerators in {product-title}, instead of relying on statically sliced GPUs defined when the node is initially started. This allows you to dynamically slice GPUs based on specific workload demands, ensuring efficient resource utilization. 

Dynamic slicing is useful in the following situations:

* You do not know all the accelerator partitions needed on every node on the cluster.

* The partition requirements change at the workload level rather than at the node level.

* You do not want to learn or use new APIs to request accelerators slices.

* You prefer to use stable device plugins APIs for creating partitions.

 :FeatureName: Dynamic Accelerator Slicer Operator

> The DAS Operator currently ships with a reference implementation for NVIDIA Multi-Instance GPU (MIG) and is designed to support additional technologies such as NVIDIA MPS or GPUs from other vendors.

Author: Changed.

The DAS Operator components interact to provide GPU slice allocation. Pods requesting GPU slices are mutated by a webhook to use the `mig.das.com` extended resource. The scheduler plugin tracks slice availability and creates `AllocationClaim` objects processed by the device plugin on each node.

.Limitations

> Keep in mind the following limitations when using the Dynamic Accelerator Slicer Operator.

Author: changed.

:_mod-docs-content-type: ASSEMBLY
[id="das-about-dynamic-accelerator-slicer-operator"]
= Dynamic Accelerator Slicer (DAS) Operator 
include::_attributes/common-attributes.adoc[]
:context: das-about-dynamic-accelerator-slicer-operator

toc::[]

[IMPORTANT]
====
Dynamic Accelerator Slicer Operator (DAS) is a Technology Preview feature and is subject to Technology Preview support limitations. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. 

For more information, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope]. 
====

> Running data intensive workloads on the platform requires efficient GPU utilization to offset the high cost of the resources involved. One approach for maximizing efficiency is through the use of oversubscription of the GPU resources by time-slicing multiple workloads. This method provides for sharing GPU resources by alternating the execution times of the workloads.

Reviewer: This is incorrect, DAS operator (formally instaslice) never used time slicing.

Author: @harche This doesn't imply that DAS uses time slicing. Rather, it refers to the use of static slicing, as the following paragraph explains and the paragraph after that talks about the advantages of dynamic slicing over time slicing. I only used those paragraphs on static slicing for comparison. 
I can see where this could be confusing. If you don't think this comparison is necessary, I'll remove lines 16-18. Let me know.

> Running data intensive workloads on the platform requires efficient GPU utilization to offset the high cost of the resources involved. One approach for maximizing efficiency is through the use of oversubscription of the GPU resources by time-slicing multiple workloads. This method provides for sharing GPU resources by alternating the execution times of the workloads.

Reviewer: ```suggestion
```

This approach allows you to define static default time-slicing configurations, but it can lead to resource wastage if the slicing does not align with the actual workload demands. Additionally, changing static partitions created at node boot-time requires the eviction of all the workloads at the node level in order to create a new set of partitions. 

> The Dynamic Accelerator Slicer (DAS) Operator allows you to optimize GPU utilization for workloads on {product-title} through the use of dynamically slicing GPU accelerators in {product-title}, instead of relying on statically sliced GPUs defined when the node is initially started. This allows you to dynamically slice GPUs based on specific workload demands, ensuring efficient resource utilization. 

Reviewer: ```suggestion
The Dynamic Accelerator Slicer (DAS) Operator allows you to optimize GPU utilization for workloads on {product-title} through the use of dynamically slicing GPU accelerators in {product-title}, instead of relying on statically sliced GPUs defined when the node is booted. This allows you to dynamically slice GPUs based on specific workload demands, ensuring efficient resource utilization. 
```

Dynamic slicing is useful in the following situations:

* You do not know all the accelerator partitions needed on every node on the cluster.

> * The partition requirements change at the workload level rather than at the node level.

Reviewer: ```suggestion
```

> * You do not want to learn or use new APIs to request accelerators slices.

Reviewer: ```suggestion
```

Author: removed.

* You prefer to use stable device plugins APIs for creating partitions.

:FeatureName: Dynamic Accelerator Slicer Operator

> The DAS Operator currently includes a reference implementation for NVIDIA Multi-Instance GPU (MIG) and is designed to support additional technologies such as NVIDIA MPS or GPUs from other vendors.

Reviewer: ```suggestion
The DAS Operator currently includes a reference implementation for NVIDIA Multi-Instance GPU (MIG) and is designed to support additional technologies such as NVIDIA MPS or GPUs from other vendors in future.
```

Author: changed - "in the future".

The DAS Operator components interact to provide GPU slice allocation. Pods requesting GPU slices are mutated by a webhook to use the `mig.das.com` extended resource. The scheduler plugin tracks slice availability and creates `AllocationClaim` objects processed by the device plugin on each node.

.Limitations

Keep in mind the following limitations when using the Dynamic Accelerator Slicer Operator.

 * You need to identify potential incompatibilities and ensure the system works seamlessly with various GPU drivers and operating systems.

 * The Operator only works with specific NVIDIA GPUs and drivers (H100, A100).

>  * When using the Operator, only the first container of the pod can access the GPU. 

Reviewer: This is no longer true for the TP release, please remove this line.

Author: Removed.

 * The Operator cannot use only a subset of the GPUs of a node.

>  * The NVIDIA GPU Operator cannot be used together with the Dynamic Accelerator Slicer Operator to manage the GPU resources of a cluster.

Author: changed.
